freqtrade/configuration.py

```diff
@@ -375,22 +376,41 @@ def get_config(self) -> Dict[str, Any]:
 
         return self.config
 
-    def check_exchange(self, config: Dict[str, Any]) -> bool:
+    def check_exchange(self, config: Dict[str, Any], check_for_bad: bool = True) -> bool:
         """
         Check if the exchange name in the config file is supported by Freqtrade
-        :return: True or raised an exception if the exchange if not supported
+        :param check_for_bad: if True, check the exchange against the list of known 'bad'
+                              exchanges
+        :return: False if exchange is 'bad', i.e. is known to work with the bot with
+                 critical issues or does not work at all, crashes, etc. True otherwise.
+                 raises an exception if the exchange if not supported by ccxt
+                 and thus is not known for the Freqtrade at all.
         """
-        exchange = config.get('exchange', {}).get('name').lower()
-        if not is_exchange_supported(exchange):
-
-            exception_msg = f'Exchange "{exchange}" not supported.\n' \
-                            f'The following exchanges are supported: ' \
-                            f'{", ".join(supported_exchanges())}'
+        logger.info("Checking exchange...")
 
-            logger.critical(exception_msg)
+        exchange = config.get('exchange', {}).get('name').lower()
+        if not is_exchange_available(exchange):
             raise OperationalException(
-                exception_msg
+                    f'Exchange "{exchange}" is not supported by ccxt '
+                    f'and therefore not available for the bot.\n'
+                    f'The following exchanges are supported by ccxt: '
+                    f'{", ".join(available_exchanges())}'
             )
 
-        logger.debug('Exchange "%s" supported', exchange)
+        logger.info(f'Exchange "{exchange}" is supported by ccxt '
+                    f'and therefore available for the bot.')
```

Comment:
I would move this message to the "else" tree below... it's only relevant if its not binance/bittrex ...

Reply:
> I would move this message to the "else" tree below... it's only relevant if its not binance/bittrex ...

So, would

`Exchange "{exchange}" is available for the bot but not officially supported by the Freqtrade development team. It may work flawlessly <...>`

be okay in that else part?

Or
`Exchange "{exchange}" is supported by ccxt and therefore available for the bot but not officially supported by the Freqtrade development team. It may work flawlessly <...>`
?

Reply:
i guess i would prefer the 2nd message - but really it's up to you (just would try to avoid having multiple messages saying the same thing for supported exchanges).

Reply:
Done, pls review.

Actually, we can now check the value returned by this function and stop working for 'bad' exchanges (main, backtest and hyperopt; util subcommands should not check for it). In this case a bool 'work_with_bad_exchanges' configuration setting can be introduced in the internals section of the config for development purposes.